dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1459,8 +1434,49 @@ class CompileTest {
         throw Exception('Unsupported option for fake devices');
       case DeviceOperatingSystem.fuchsia:
         throw Exception('Unsupported option for Fuchsia devices');
+      case DeviceOperatingSystem.ios:
       case DeviceOperatingSystem.macos:
-        throw Exception('Unsupported option for macOS devices');
+        unawaited(stderr.flush());
+        late final String deviceId;
```

Comment:
this can actually be `final` because of definite assignment analysis https://dart.dev/null-safety/understanding-null-safety#definite-assignment-analysis